Core/KSPManager.cs

```diff
@@ -151,6 +152,103 @@ public KSP AddInstance(KSP ksp_instance)
+
+                // Add the new instance to the registry
+                KSP new_instance = new KSP(new_path, new_name, User);
+                AddInstance(new_instance);
+            }
+            // Thrown by AddInstance() if created instance is not valid.
+            // Thrown f.e. if something went wrong with copying.
+            catch (NotKSPDirKraken kraken)
+            {
+                throw kraken;
+            }
+            // Thrown by CopyDirectory() if directory doesn't exist. Shouldn't be thrown anytime.
+            catch (DirectoryNotFoundKraken kraken)
+            {
+                throw kraken;
+            }
+            // Thrown by CopyDirectory() if the specified folder already exists and is not empty.
+            catch (IOException e)
+            {
+                throw e;
+            }
+        }
+
+        /// <summary>
+        /// Create a new fake KSP instance
+        /// </summary>
+        /// <param name="new_name">The name for the new instance.</param>
+        /// <param name="new_path">The loaction of the new instance.</param>
+        /// <param name="version">The version of the new instance. Should have a build number.</param>
+        /// <param name="DLC">Whether to fake the DLC too.</param>
+        /// <param name="dlcVersion">The version of the DLC. Can be null if DLC == false.</param>
+        public void FakeInstance(string new_name, string new_path, KspVersion version, bool DLC, string dlcVersion)
+        {
+            if (!version.IsValid())
+            {
+                throw new ArgumentOutOfRangeException(nameof(version), "The specified KSP version is not a valid version.");
+            }
+            if (Directory.Exists(new_path) && (Directory.GetFiles(new_path).Length != 0 || Directory.GetDirectories(new_path).Length != 0))
+            {
+                throw new BadInstallLocationKraken("The specified folder already exists and is not empty.");
+            }
+
+            try
+            {
+                log.DebugFormat("Creating folder structure and text files at {0} for KSP version {1}", Path.GetFullPath(new_path), version.ToString());
+
+                // Create a KSP root directory, containing a GameData folder, a buildID.txt/buildID64.txt and a readme.txt
+                Directory.CreateDirectory(new_path);
+                Directory.CreateDirectory(Path.Combine(new_path, "GameData"));
+
+                // Don't write the buildID.txts if we have no build, otherwise it would be -1.
+                if (version.IsBuildDefined)
+                {
+                    File.WriteAllText(Path.Combine(new_path, "buildID.txt"), String.Format("build id = {0}", version.Build));
+                    File.WriteAllText(Path.Combine(new_path, "buildID64.txt"), String.Format("build id = {0}", version.Build));
+                }
+
+                File.WriteAllText(Path.Combine(new_path, "readme.txt"), String.Format("Version {0}", version.ToString()));
```

Comment:
Will this output the build ID? SQUAD's readme.txt file doesn't contain that.

Reply:
Yep, it does, but it shouldn't. Fixing...